### PR TITLE
[SYCL][NATIVECPU] Avoid overwriting offload kind on Native CPU

### PR DESCRIPTION
**Pull request by PietroGhg**
For Native CPU, the host and the device triples are the same. This means that the cache entries for `BuildJobsForAction` are distinguished only by the offload kind (`-host` or `-sycl`). Since the offload kind is being overwritten when processing `OffloadUnbundlingJobAction`, this lead to cache entries being overwritten on Native CPU, with corresponding build failures.

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -9091,7 +9091,8 @@ InputInfoList Driver::BuildJobsForActionNoCache(
       // in this instance
       Action::OffloadKind DependentOffloadKind;
       if (UI.DependentOffloadKind == Action::OFK_SYCL &&
-          TargetDeviceOffloadKind == Action::OFK_None)
+          TargetDeviceOffloadKind == Action::OFK_None &&
+          !isSYCLNativeCPU(C.getDefaultToolChain().getTriple(), TC->getTriple()))
```
Comment (mdtoguchi) [inline clang/lib/Driver/Driver.cpp:9095]: Is there a test that can be updated to track the kind consistency?

Comment (PietroGhg) [reply clang/lib/Driver/Driver.cpp:9095]: I don't think we have one, I'll add it to the PR

Comment (PietroGhg) [reply clang/lib/Driver/Driver.cpp:9095]: @mdtoguchi I've updated the lit test for native cpu, I'm still looking into the CI failures in the driver lit tests

**Review by uwedolinsky — APPROVED** [review]
This fix looks good to me for now. We will revisit querying for Native CPU to improve it in one of the next PRs.

**Review by mdtoguchi — APPROVED** [review]

**Comment by PietroGhg** [comment]
Hello @intel/llvm-gatekeepers, this looks ready to be merged, thank you

**Pull request merged**